src/agent/src/policy.rs

```diff
@@ -120,13 +154,56 @@ impl AgentPolicy {
         let query = format!("data.agent_policy.{ep}");
         self.engine.set_input_json(ep_input)?;
 
-        let mut allow = match self.engine.eval_bool_query(query, false) {
-            Ok(a) => a,
-            Err(e) => {
-                if !self.allow_failures {
-                    return Err(e);
+        let results = self.engine.eval_query(query, false)?;
```

Comment:
Consider using [Engine::eval_rule](https://docs.rs/regorus/latest/regorus/struct.Engine.html#method.eval_rule) if the query is always path to a rule. `eval_rule` returns a `Value` directly.

Reply:
Thanks! Will tackle this on a separate PR